tests/test_file_encoding.py

```diff
@@ -0,0 +1,33 @@
+import re
+from pathlib import Path
+
+
+def _no_encoding_on_file_open(filepath: str):
+    r"""Find all instances where a non-binary file is opened without UTF-8 encoding.
+
+    This function uses regular expressions to find instances where Python's `open()` function is used to open
+    non-binary files. See below for an explanation of the regular expression:
+
+    (?!.*\b(?:encoding|rb|w|wb|w+|wb+|ab|ab+)\b): Lookahead and discard match if `encoding` or `rb` etc are
+    arguments of `open()`.
+
+    (?<=\s): Lookbehind and match if `open()` predeceded by one whitespace.
+
+    (open)\((.*)\): Capture everything in parentheses of `open()`.
+    """
+
+    with open(filepath, "r", encoding="utf-8") as input_file:
+        regexp = re.compile(r"(?!.*\b(?:encoding|rb|w|wb|w+|wb+|ab|ab+)\b)(?<=\s)(open)\((.*)\)")
```

Comment:
Can you update the regex with your latest version ?

Reply:
Sorry if I wasn't clear in my previous messages - the simplified regex (amended in 
commit 03f98c66e8d9b3cea79ca10f4b220dab9f7e9e99) _is_ the latest version. 

Your suggestion allowed me to replace ugly patterns like `(?<!csv.reader\()` and `(?<!\.)` with just a whitespace lookbehind `(?<=\s)`.

Please let me know if I misunderstood your comment!

Reply:
Oh ok I see it's good then